Test/diffs/difftester.py

```diff
@@ -0,0 +1,955 @@
+#!/usr/bin/env python3
```

Comment:
Could you structure the code into multiple files, to make it easier to digest? For example, a separate file for each supported front-end might help, or for each supported type of input.

Reply:
I have mixed feelings about this: there's usefulness to having everything in one file.  I agree that 1000 lines is getting close to the manageable limit.  Happy to refactor if you think the benefits overweight the downsides.

Reply:
If everything is in one file, how do I see the hierarchical structure of the code? Maybe an IDE with an outline view can show the classes in the file, but that's not as accessible as having the files define that structure. What's the benefit of a single file?

Reply:
> If everything is in one file, how do I see the hierarchical structure of the code?

You search for classes; there's 10 to 20 of them in here, so that remains manageable.  The advantage of a single file is that you don't need an IDE to navigate around; all relevant definitions are in the same place, so you can just search around.

Reply:
I personally prefer a big file than many small ones, because indeed it makes it easier to search around for the concepts, especially in a script that is doing only one thing and where we don't need to share code outside.

However, to keep it readable, since the classes might be for different purposes, I would advise to add a comment on the top of the file explaining what are all the classes that were defined and what purpose they have, so that it will make it easier to maintain down the road.

Reply:
FWIW, I have a slightly different perspective:  I think the cost of rewriting in C# is not huge, so I can do it if we find it useful.  It's probably a day of two at most.  But how about we first give this script a bit of a try, and if it turns out to be actually useful, we can decide whether to invest more in it.

Reply:
I agree with this.

Reply:
Alright. Let's see ^^

Reply:
> But how about we first give this script a bit of a try, and if it turns out to be actually useful, we can decide whether to invest more in it.

What do you mean by giving it a try? If we turn it on in the build-server then we're using it right? What would be a reason to undo that then?

I think normally we assume checked in code is there to stay, and the time of merging is it also the right time to optimise it for readability. In my opinion, having code in anything other than the primary language of a codebase makes it much less readable.

Already for this PR I don't feel as capable as a reviewer as I would for a C# PR.